distro/mqtt.go

```diff
@@ -23,14 +25,38 @@ type mqttSender struct {
 
 // NewMqttSender - create new mqtt sender
 func NewMqttSender(addr models.Addressable) Sender {
+
+	protocol := strings.ToLower(addr.Protocol)
+
 	opts := MQTT.NewClientOptions()
-	broker := "tcp://" + addr.Address + ":" + strconv.Itoa(addr.Port)
+	broker := protocol + "://" + addr.Address + ":" + strconv.Itoa(addr.Port) + addr.Path
 	opts.AddBroker(broker)
 	opts.SetClientID(addr.Publisher)
 	opts.SetUsername(addr.User)
 	opts.SetPassword(addr.Password)
 	opts.SetAutoReconnect(false)
 
+	if protocol == "tcps" ||
+		protocol == "ssl" ||
+		protocol == "tls" {
+
+		cert, err := tls.LoadX509KeyPair(cfg.MQTTSCert, cfg.MQTTSKey)
+
+		if err != nil {
+			logger.Fatal("Failed loading x509 data")
```

Comment:
Fatal shouldn't be used. Change to error or warning. registrationInfo.update should also be fixed for NewMqttSender returning nil.